File: Data Structures & Algorithms/islands-and-treasure/submission-4.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        
        ROWS, COLS = len(grid), len(grid[0])
        visit = set()
        q = deque()

        def adjustCell(r,c):

                if ( r < 0 or c < 0 or r >= ROWS or c >= COLS
                    or (r,c) in visit or grid[r][c] == -1):
                    return
                else:
                    visit.add((r,c))
                    q.append((r,c))
                    return

        for r in range(ROWS):
            for c in range(COLS):
                if grid[r][c] == 0:
                    visit.add((r,c))
                    q.append((r,c))

        dist = 0
        while q:
            for i in range(len(q)):
                r,c = q.popleft()
                grid[r][c] = dist
                adjustCell(r+1,c)
                adjustCell(r-1,c)
                adjustCell(r,c-1)
                adjustCell(r,c+1)
            dist += 1
            

File: Data Structures & Algorithms/search-2d-matrix/submission-4.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {

        int rows = matrix.length;
        int cols = matrix[0].length;

        int possibleRow = 0;
        for ( int i = 0; i < rows; i++ ) {
            if ( matrix[i][cols - 1] < target ) continue;
            else if ( matrix[i][cols - 1] >= target ) {
                possibleRow = i;
                break;
            } 
        }

        for ( int i = 0; i < cols; i++ ) {
            if (matrix[possibleRow][i] == target) return true;
        }
        return false;
        
    }
}
